### ISSUE TITLE
pre-processing CityScapes dataset google drive

### ISSUE THREAD

**Issue by TianhaoFu** [lorenmt/mtan#40]
Hi,
Thanks for your code!

Could you please provide a **pre-processing CityScapes dataset google dive link**? 
I cannot download from dropbox.

Thanks!

**Comment by lorenmt** [comment]
Could you specify what problem did you encounter? The dropbox link is still valid.

**Comment by TianhaoFu** [comment]
Thanks for your reply!
My problem is that the raw dataset file is 2.4GB. But My dropbox space is only 2 GB because I 'm not the paying user. So I cannot download the dataset.

BTW, could you provide your cityscape dataloader file?
Thanks:)

**Comment by lorenmt** [comment]
You can download the data without having a dropbox account. Check out the download page in detail, there must be a direct download link.

I am sorry that I am not going to provide the cityscapes dataloader, you can follow my implementation on nyuv2 dataloader and code it by yourself, it should be quite straightforward. :D

**Comment by TianhaoFu** [comment]
> You can download the data without having a dropbox account. Check out the download page in detail, there must be a direct download link.
> 
> I am sorry that I am not going to provide the cityscapes dataloader, you can follow my implementation on nyuv2 dataloader and code it by yourself, it should be quite straightforward. :D

thanks!

Did you do any data augmentatioin on cityscapes ? 
Or you only use raw 5000 pics to train and test?
:)

**Comment by lorenmt** [comment]
In the original paper, no, I did not apply data augmentation for both Cityscapes and NYUv2.

But I would suggest applying augmentation to be consistent with the most recent benchmark on MTL.

You can take a look at this paper: https://arxiv.org/abs/2004.13379 and follow their preprocessing step.

**Comment by TianhaoFu** [comment]
> https://arxiv.org/abs/2004.13379

thanks. 
In your paper you You mentioned that when you computing depth loss, you  _inverse depth in CityScapes outdoor scene dataset as standard_ , So **in your cityscape dataset your depth is whether raw depth data or inversed depth data ?**

And **in cityscape dataset did you compute loss and evaluation metric follow the code in your respository as follows**?
```
if task_type == 'depth':
        # depth loss: l1 norm
        loss = torch.sum(torch.abs(x_pred - x_output) * binary_mask) / torch.nonzero(binary_mask, as_tuple=False).size(0)
def depth_error(x_pred, x_output):
    device = x_pred.device
    binary_mask = (torch.sum(x_output, dim=1) != 0).unsqueeze(1).to(device)
    x_pred_true = x_pred.masked_select(binary_mask)
    x_output_true = x_output.masked_select(binary_mask)
    abs_err = torch.abs(x_pred_true - x_output_true)
    rel_err = torch.abs(x_pred_true - x_output_true) / x_output_true
    return (torch.sum(abs_err) / torch.nonzero(binary_mask, as_tuple=False).size(0)).item(), \
           (torch.sum(rel_err) / torch.nonzero(binary_mask, as_tuple=False).size(0)).item()

```
thank u very much!

**Comment by lorenmt** [comment]
The CityScapes dataset in my dropbox is inverse depth, you can easily verify that by noticing all values in depth data are in the range (0, 1).

The metrics in my repo are universal and standard, you can directly apply them on any datasets.

There is one issue however I have mentioned in my readme's latest update: The original mIoU computation is to average IoU in each image, which is okay, but to make it more consistent with the standard segmentation benchmark, I revised it into average IoU in the entire dataset (accumulate pixel prediction across all images). The original mIoU implementation is still in the file but commented. So I am afraid you have to rerun all baselines to have a fair comparison.

**Comment by TianhaoFu** [comment]
> The CityScapes dataset in my dropbox is inverse depth, you can easily verify that by noticing all values in depth data are in the range (0, 1).
> 
> The metrics in my repo are universal and standard, you can directly apply them on any datasets.
> 
> There is one issue however I have mentioned in my readme's latest update: The original mIoU computation is to average IoU in each image, which is okay, but to make it more consistent with the standard segmentation benchmark, I revised it into average IoU in the entire dataset (accumulate pixel prediction across all images). The original mIoU implementation is still in the file but commented. So I am afraid you have to rerun all baselines to have a fair comparison.

Thanks a lot!

btw, what the data asytpe in your cityscape dataset?
np.float32? or np.float64?

thanks!

**Comment by lorenmt** [comment]
No. The difference would be minimal.

**Comment by TianhaoFu** [comment]
> No. The difference would be minimal.

Hi, Coule tell me what kind of method/code you used to resize the cityscape dataset to [128,256]?

Thanks. @lorenmt

**Comment by lorenmt** [comment]
Simply use PIL.resize would do the work.
